pkg/hintrunner/zero/zerohint_ec_test.go

```diff
@@ -418,5 +418,113 @@ func TestZeroHintEc(t *testing.T) {
 				check: consecutiveVarValueEquals("res", []*fp.Element{feltString("2"), feltString("2"), &utils.FeltZero}),
 			},
 		},
+		"FastEcAddAssignNewY": {
+			{
+				operanders: []*hintOperander{},
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					ctx.ScopeManager.EnterScope(map[string]any{})
+
+					slopeBig := big.NewInt(100)
+					x0Big := big.NewInt(20)
+					new_xBig := big.NewInt(10)
+					y0Big := big.NewInt(10)
+
+					err := ctx.ScopeManager.AssignVariable("slope", slopeBig)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
+					err = ctx.ScopeManager.AssignVariable("x0", x0Big)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
+					err = ctx.ScopeManager.AssignVariable("new_x", new_xBig)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
+					err = ctx.ScopeManager.AssignVariable("y0", y0Big)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
```

Comment:
You can pass all these variables at the creation of the new scope, that's the idea of passing a map as an argument in `EnterScope` function. I think this would make the code look a lot smaller and simpler

Reply:
Ah good idea!